linked_list.rb

```diff
@@ -19,77 +19,243 @@ def initialize
   # method to add a new node with the specific data value in the linked list
   # insert the new node at the beginning of the linked list
   def insert(value)
-    puts "Not implemented"
+    new_node = Node.new(value)
+
+    if @head == nil
+      @head = new_node
+    else
+      new_node.next = @head
+      @head = new_node
+    end
   end
 
   # method to find if the linked list contains a node with specified value
   # returns true if found, false otherwise
   def search(value)
-    puts "Not implemented"
+    current = @head
+    until current == nil
+      if current.data == value
+        return true
+      end
+      current = current.next
+    end
+    return false
   end
 
   # method to return the max value in the linked list
   # returns the data value and not the node
   def find_max
-    puts "Not implemented"
+    current = @head
+    max = current

```

Comment:
What is head is nil? You won't be able to return max.data in that case. Always check for nil before dereferencing.